test-harness/src/test/java/com/netflix/conductor/tests/integration/AbstractWorkflowServiceTest.java

```diff
@@ -5350,11 +5337,28 @@ public void testTerminateTaskWithFailedStatus() {
         terminateWorkflowTask.setInputParameters(terminateTaskInputParams);
         terminateWorkflowTask.setTaskReferenceName("terminate0");
 
-        WorkflowTask workflowTask2 = new WorkflowTask();
-        workflowTask2.setName("junit_task_2");
-        workflowTask2.setTaskReferenceName("t2");
+        // right fork
+        WorkflowTask subWorkflow = new WorkflowTask();
+        subWorkflow.setType(SUB_WORKFLOW.name());
+        SubWorkflowParams sw = new SubWorkflowParams();
+        sw.setName(subworkflowDef.getName());
+        subWorkflow.setSubWorkflowParam(sw);
+        subWorkflow.setTaskReferenceName("sw1");
+
+        fanoutTask.getForkTasks().add(Arrays.asList(subWorkflow));
+        fanoutTask.getForkTasks().add(Arrays.asList(lambdaWorkflowTask, waitWorkflowTask, terminateWorkflowTask));
+
+        // join task
+        WorkflowTask joinTask = new WorkflowTask();
+        joinTask.setType(TaskType.JOIN.name());
+        joinTask.setTaskReferenceName("fanouttask_join");
+        joinTask.setJoinOn(Arrays.asList("sw1", "leftfork-wait0"));
```

Comment:
The last task in the left fork is the terminate task, however, the join is set to an intermediate task (wait). This looks incorrect.

Reply:
made the requested change